rust-libretro/src/environment.rs

```diff
@@ -979,17 +1045,19 @@ pub unsafe fn set_hw_shared_context(callback: retro_environment_t) -> bool {
 pub fn get_vfs_interface(
     callback: retro_environment_t,
     data: retro_vfs_interface_info,
-) -> Option<retro_vfs_interface_info> {
+) -> Result<retro_vfs_interface_info, EnvironmentCallError> {
```

Comment:
The only required input to `RETRO_ENVIRONMENT_GET_VFS_INTERFACE` is the version number, so I'd argue that you could simplify the use of this function by only accepting a version number.

Reply:
`SetEnvironmentContext::enable_vfs_interface(min_version)` provides a wrapper around this function that also stores the struct in the `CoreWrapper` for you and enables functions like `GenericContext::vfs_get_path()` to work.
So I don’t think that changing the API is necessary here.